astropy/units/physical.py

```diff
@@ -367,17 +366,9 @@ def __str__(self) -> str:
         return "/".join(self._physical_type)
 
     @staticmethod
-    def _dimensionally_compatible_unit(
-        obj: object,
-    ) -> core.UnitBase | NotImplementedType:
+    def _dimensionally_compatible_unit(obj: object) -> core.UnitBase | None:
         """
         Return a unit that corresponds to the provided argument.
-
-        If a unit is passed in, return that unit.  If a physical type
-        (or a `str` with the name of a physical type) is passed in,
-        return a unit that corresponds to that physical type.  If the
-        number equal to ``1`` is passed in, return a dimensionless unit.
-        Otherwise, return `NotImplemented`.
```

Comment:
I would have had to edit the last line anyways, but then I noticed that the first sentence is not always true and anybody who wants to know what this method does is better off reading the code itself anyways.